examples/stream/port_duplex.js

```diff
@@ -4,35 +4,37 @@ const { Duplex } = require('stream');
 
 class PortDuplex extends Duplex {
   #port = undefined;
-  #transfer = false;
 
   constructor (port, options) {
     const {
       readable = true,
-      writable = true,
-      transfer = false
+      writable = true
     } = { ...options };
     if (typeof readable !== 'boolean') {
       throw new TypeError('readable must be a boolean');
     }
     if (typeof writable !== 'boolean') {
       throw new TypeError('writable must be a boolean');
     }
-    if (typeof transfer !== 'boolean') {
-      throw new TypeError('transfer must be a boolean');
-    }
     super({ autoDestroy: true, readable, writable });
     this.#port = port;
-    this.#transfer = transfer;
     this.#port.onmessage = PortDuplex.#onmessage.bind(this);
   }
 
   _write (chunk, encoding, callback) {
     if (typeof chunk === 'string') {
       chunk = Buffer.from(chunk, encoding);
     }
-    const transferList = this.#transfer ? [chunk.buffer] : undefined;
-    this.#port.postMessage(chunk, transferList);
+    // Be sure to always copy the chunk here and never use a
+    // transferList. There are several reasons:
+    // a) Buffer instances are most often created off a pool
+    //    and share the same underlying common ArrayBuffer,
+    //    transferring those can break Node.js in many ways.
```

Comment:
See the comment in the Node.js thread, this is true but not necessarily a concern in the future. The sceond concern is definitely there, though :)